pandas/core/resample.py

```diff
@@ -1540,20 +1541,39 @@ def _get_period_bins(self, ax):
                 data=[], freq=self.freq, name=ax.name)
             return binner, [], labels
 
+        freq_mult = self.freq.n
+
         start = ax.min().asfreq(self.freq, how=self.convention)
         end = ax.max().asfreq(self.freq, how='end')
+        offset = 0
+
+        # GH 23882
+        if self.base:
+            # get base adjusted bin edge labels
+            p_start, end = _get_period_range_edges(start,
+                                                   end,
+                                                   self.freq,
+                                                   closed=self.closed,
+                                                   base=self.base)
+
+            # Get offset for bin edge (not label edge) adjustment
```

Comment:
why is this still needed? shouldn't this be in get_period_range_edges?

Reply:
See the last paragraph of this comment. 
https://github.com/pandas-dev/pandas/pull/23941#discussion_r237332194

Essentially, get_range_edges only accounts for the resultant labels, not the actual bins being aggregated. So the bins edges need to be shifted to reflect the base parameter.